src/mint.rs

```diff
@@ -56,9 +58,10 @@ impl Mint {
         vec![tx]
     }
 
-    pub fn create_entries(&self) -> Vec<Entry> {
+    pub fn create_entries(&mut self) -> Vec<Entry> {
         let e0 = Entry::new(&self.seed(), 0, vec![], false);
         let e1 = Entry::new(&e0.id, 0, self.create_transactions(), false);
+        self.last_id = e1.id;
```

Comment:
B/c of the element of randomness on contract ids, create_entries no longer generates predictable entries (which breaks quite a few tests). Instead of this fix here, we could also configure the Contract constructor to take an input id as an option, and pass that to the Transaction::new() if we want to be able to reproduce consistent copies of Contracts.